cli/commands/benchmark-web-vitals.mjs

```diff
@@ -514,124 +514,141 @@ async function benchmarkURL( url, metricsDefinition, params, logProgress ) {
-		if ( params.cpuThrottleFactor ) {
-			await page.emulateCPUThrottling( params.cpuThrottleFactor );
-		}
+		/** @type {Browser} */
+		let browser;
 
-		if ( params.networkConditions ) {
-			await page.emulateNetworkConditions( params.networkConditions );
-		}
+		try {
+			browser = await launchBrowser();
+			if ( logProgress ) {
+				logPartial(
+					`Benchmarking ${ requestNum + 1 } / ${ params.amount }...`
+				);
+			}
+			const page = await browser.newPage();
+			await page.setBypassCSP( true ); // Bypass CSP so the web vitals script tag can be injected below.
+			if ( params.cpuThrottleFactor ) {
+				await page.emulateCPUThrottling( params.cpuThrottleFactor );
+			}
 
-		if ( params.emulateDevice ) {
-			await page.emulate( params.emulateDevice );
-		}
-		if ( params.windowViewport ) {
-			await page.setViewport( {
-				...( params.emulateDevice
-					? params.emulateDevice.viewport
-					: {} ),
-				...params.windowViewport,
-			} );
-		}
+			if ( params.networkConditions ) {
+				await page.emulateNetworkConditions( params.networkConditions );
+			}
 
-		// Load the page.
-		const urlObj = new URL( url );
-		urlObj.searchParams.append( 'rnd', String( Math.random() ) );
+			if ( params.emulateDevice ) {
+				await page.emulate( params.emulateDevice );
+			}
+			if ( params.windowViewport ) {
+				await page.setViewport( {
+					...( params.emulateDevice
+						? params.emulateDevice.viewport
+						: {} ),
+					...params.windowViewport,
+				} );
+			}
 
-		// Make sure any username and password in the URL is passed along for authentication.
-		if ( urlObj.username && urlObj.password ) {
-			await page.authenticate( {
-				username: urlObj.username,
-				password: urlObj.password,
-			} );
-		}
+			// Load the page.
+			const urlObj = new URL( url );
+			urlObj.searchParams.append( 'rnd', String( Math.random() ) );
 
-		const response = await page.goto( urlObj.toString(), {
-			waitUntil: 'networkidle0',
-		} );
-		if ( scriptTag ) {
-			await page.addScriptTag( { content: scriptTag, type: 'module' } );
-		}
+			// Make sure any username and password in the URL is passed along for authentication.
+			if ( urlObj.username && urlObj.password ) {
+				await page.authenticate( {
+					username: urlObj.username,
+					password: urlObj.password,
+				} );
+			}
 
-		if ( response.status() !== 200 ) {
-			if ( logProgress ) {
-				log(
-					formats.error(
-						`Error: Bad response code ${ response.status() }.`
-					)
-				);
+			const response = await page.goto( urlObj.toString(), {
+				waitUntil: 'networkidle0',
+			} );
+			if ( scriptTag ) {
+				await page.addScriptTag( {
+					content: scriptTag,
+					type: 'module',
+				} );
 			}
-			continue;
-		}
 
-		completeRequests++;
+			if ( response.status() !== 200 ) {
+				if ( logProgress ) {
+					log(
+						formats.error(
+							`Error: Bad response code ${ response.status() }.`
+						)
+					);
+				}
+				continue;
```

Comment:
Since you're adding a central error logger in the `catch` now, it would be great to `throw` a new error here with the message instead of logging directly. Just to centralize / remove duplicated logging code :)

Reply:
Good idea. Done in 7fe49f02a8be9e820ee0985d25181c0875fbc06d